addons/netfox.extras/state-machine/network-state-machine.gd

```diff
@@ -0,0 +1,51 @@
+@icon("res://addons/netfox/icons/state-synchronizer.svg")
+extends Node
+class_name NetworkedStateMachine
```

Comment:
I think we could add a logger to this class too, see [NetworkTime._logger] as an example. You can use [`_NetfoxLogger.for_extras`] to instantiate the logger. Using it over `push_warning` can be a bit more useful, as it logs current tick and peer id too. 

Also useful for debugging during development.

[NetworkTime._logger]: https://github.com/foxssake/netfox/blob/a557f98aa8ed49825fe55668a210b51b81ad7e01/addons/netfox/network-time.gd#L275
[`_NetfoxLogger.for_extras`]: https://github.com/foxssake/netfox/blob/a557f98aa8ed49825fe55668a210b51b81ad7e01/addons/netfox.internals/logger.gd#L44